python/paddle/fluid/tests/unittests/test_view_op_reuse_allocation.py

```diff
@@ -0,0 +1,110 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+import unittest
+
+import numpy as np
+
+from op_test import OpTest
+import paddle
+
+
+# NOTE(pangyoki): Tensor View Strategy.
+# Refer to `op_function_generator.py`.
+# For view op, a new output varbase will be created, and this varbase will
+# reuse the input varbase's allocation.
+# View APIs include: `squeeze`, `unsqueeze`, `reshape`, `flatten`, `detach`
+class TestDygraphViewReuseAllocation(unittest.TestCase):
+    def view_api_processing(self, var):
+        return paddle.squeeze(var)
+
+    def get_numpy_result(self, var_numpy):
+        return var_numpy.squeeze()
+
+    def test_view_api(self):
+        var = paddle.rand([2, 3, 1])
+        view_var = self.view_api_processing(var)
+        view_var[0] = 2.
+        self.assertNotEqual(var.shape, view_var.shape)
```

Comment:
Better to check the right shape, too. For example:
```
self.assertEqual(view_var.shape, [2,3])
```

Reply:
done